osu.Game.Tests/Visual/Gameplay/TestSceneAutoplay.cs

```diff
@@ -24,25 +22,31 @@ protected override Player CreatePlayer(Ruleset ruleset)
 
         protected override void AddCheckSteps()
         {
-            AddUntilStep("score above zero", () => ((ScoreAccessiblePlayer)Player).ScoreProcessor.TotalScore.Value > 0);
-            AddUntilStep("key counter counted keys", () => ((ScoreAccessiblePlayer)Player).HUDOverlay.KeyCounter.Children.Any(kc => kc.CountPresses > 2));
-            AddStep("rewind", () => track.Seek(-10000));
-            AddUntilStep("key counter reset", () => ((ScoreAccessiblePlayer)Player).HUDOverlay.KeyCounter.Children.All(kc => kc.CountPresses == 0));
+            ScoreAccessiblePlayer scoreAccessiblePlayer = null;
+
+            AddUntilStep("player loaded", () => (scoreAccessiblePlayer = (ScoreAccessiblePlayer)Player) != null);
+            AddUntilStep("score above zero", () => scoreAccessiblePlayer.ScoreProcessor.TotalScore.Value > 0);
+            AddUntilStep("key counter counted keys", () => scoreAccessiblePlayer.HUDOverlay.KeyCounter.Children.Any(kc => kc.CountPresses > 2));
+            AddStep("seek to break time", () => scoreAccessiblePlayer.GameplayClockContainer.Seek(scoreAccessiblePlayer.BreakOverlay.Breaks.First().StartTime));
+            AddUntilStep("wait for seek to complete", () =>
+                scoreAccessiblePlayer.HUDOverlay.Progress.ReferenceClock.CurrentTime >= scoreAccessiblePlayer.BreakOverlay.Breaks.First().StartTime);
+            AddAssert("test keys not counting", () => !scoreAccessiblePlayer.HUDOverlay.KeyCounter.IsCounting);
```

Comment:
I'm not sure this does much. Sure, on the surface it looks correct as it tests the flag, but *the flag itself isn't tested* (as in, it's not tested that the key counter doesn't actually increment keys with `IsCounting = false` - what if the logic checking for `IsCounting` regresses?)

Reply:
Not sure why I assumed this was already tested in KeyCounter